src/main.rs

```diff
@@ -540,3 +547,180 @@ mod provision {
         Ok(())
     }
 }
+
+mod ok {
+    use super::*;
+    use colorful::*;
+    use std::arch::x86_64;
+    use std::mem::transmute;
+    use std::str::from_utf8;
+
+    #[derive(StructOpt)]
+    pub enum SevGeneration {
+        #[structopt(about = "AMD SEV Extensions")]
+        Sev,
+
+        #[structopt(about = "AMD SEV-ES Extensions")]
+        Es,
+    }
+
+    struct CpuId {
+        pub name: &'static str,
+        pub leaf: u32,
+        pub func: fn(x86_64::CpuidResult) -> (bool, Option<String>),
+    }
+
+    struct OkTestResult {
```

Comment:
I don't think the `Ok` prefix adds anything to this type. Would you consider removing the `Ok` prefix?